cli/src/pcluster/aws/aws_resources.py

```diff
@@ -247,15 +249,24 @@ class FsxFileSystemInfo:
     def __init__(self, file_system_data):
         self.file_system_data = file_system_data
 
+    @property
+    def file_system_type(self):
+        """Return the type of FSx file system (LUSTRE, WINDOWS, ONTAP, or OPENZFS). WINDOWS is not supported."""
+        return self.file_system_data.get("FileSystemType")
+
     @property
     def mount_name(self):
         """Return MountName of the filesystem."""
-        return self.file_system_data.get("LustreConfiguration").get("MountName")
+        return (
+            self.file_system_data.get("LustreConfiguration").get("MountName")
+            if self.file_system_type == FSX_LUSTRE
+            else ""
+        )
 
     @property
     def dns_name(self):
         """Return DNSName of the filesystem."""
-        return self.file_system_data.get("DNSName")
+        return self.file_system_data.get("DNSName") if self.file_system_type in [FSX_LUSTRE, FSX_OPENZFS] else ""
```

Comment:
Done

Reply:
Thanks for adding a comment.
You say "Ontap has DNS name on storage virtual machines.".
So, we are not retrieving DNS name for Ontap here because in terms of clean coding is not correct to retrieve it from here, or because we do not need to retrieve it at all?

Reply:
The DNS name is needed. But for clean coding, we need to get the DNS name form storage virtual machine

Reply:
Questions:
1. What is the expected content of `file_system_data.get("DNSName")` for ONTAP?
1. What about adding the DNSName also for ONTAP within `file_system_data` before passing it to the constructor of `FsxFileSystemInfo`?

Reply:
1. There is no such key in the response. So the result is None
2. the DNSName of Ontap is on the storage virtual machine. Why adding it to file system data?